src/test/java/com/alkemy/ong/web/controller/MemberControllerTest.java

```diff
@@ -0,0 +1,290 @@
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static com.alkemy.ong.web.controller.MemberController.*;
+import static org.hamcrest.Matchers.*;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class MemberControllerTest {
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @MockBean
+    MemberRepository memberRepository;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Test
+    void save_success() throws Exception {
+        MemberDTO memberDTO = buildDto();
+        MemberEntity entityWithoutId = buildEntityWithoutId();
+        MemberEntity resultEntityWhitId = buildEntityWhitId(1L);
+
+        when(memberRepository.save(entityWithoutId)).thenReturn(resultEntityWhitId);
+
+        mockMvc.perform(post("/members")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(memberDTO)))
+                .andExpect(status().isCreated())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.id", is(1)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.name", is("TestName")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.facebookUrl", is("https://www.facebook.com/profile")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.instagramUrl", is("https://www.instagram.com/profile")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.image", is("user/img/photo.jpg")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.description", is("Member description")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.createdAt", is("2022-03-05")));
+    }
+
+    @Test
+    void save_BadRequest() throws Exception {
+        MemberDTO memberDTO = MemberDTO.builder()
+                .name(null)
+                .image(null)
+                .build();;
+        MemberEntity entityWithoutId = buildEntityWithoutId();
+        MemberEntity resultEntityWhitId = buildEntityWhitId(1L);
+
+        when(memberRepository.save(entityWithoutId)).thenReturn(resultEntityWhitId);
+
+        mockMvc.perform(post("/members")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(memberDTO)))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void save_NotFound() throws Exception {
```

Comment:
no hay chance q en un post se encuentre un not found, esto aplica para findById, update o delete, pero no para create. No tiene sentido este test.